### PR TITLE
#3611 Remove go-couchbase bucket

### LINKED ISSUE #3611: Remove CouchbaseBucket (go-couchbase) bucket implementation
All functionality should have been moved over to gocb/cbdatasource at this point, so we'd like to remove the CouchbaseBucket code to clean up the SG code base.

Include an audit of the sg-bucket interface and remove any interfaces that are specific to go-couchbase/CouchbaseBucket.

### PR DESCRIPTION
**Pull request by bbrks**
Fixes #3611 - Removes go-couchbase bucket

### CODE REVIEW

**Comment by bbrks** [comment]
- [ ] Still awaiting integration suite run

**Comment by bbrks** [comment]
Tests passed: http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/759

File: base/bucket.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ const (
 )
 
 const (
-	GoCouchbase            CouchbaseDriver = iota
+	_                      CouchbaseDriver = iota
```
Comment (adamcfraser) [inline base/bucket.go]: Can remove this, otherwise looks good.

Comment (bbrks) [reply base/bucket.go]: done

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**